src/command/upload.rs

```diff
@@ -454,5 +467,13 @@ impl BindleConnectionInfo {
         let allow_insecure = args.is_present(ARG_INSECURE);
         args.value_of(ARG_BINDLE_URL)
             .map(|base_url| Self::new(base_url, allow_insecure))
+            .map(|mut me| {
```

Comment:
Nit: It seems like it would be simpler just to add a couple of option string args to `BindleConnectionInfo::new`?  Propagating the Some/None args into the BCI fields is what we end up doing anyway...!

Reply:
If you're okay with `Option<>` args, I totally agree on that.